lib/dialects/mssql/query-generator.js

```diff
@@ -782,101 +804,40 @@ class MSSQLQueryGenerator extends AbstractQueryGenerator {
-         * 2. The next query handles the LIMIT and OFFSET behavior by getting
-         *    the TOP N rows of the query where the row number is > OFFSET
-         * 3. The innermost query is the actual set we want information from
-         */
-        const fragment = `SELECT TOP 100 PERCENT ${attributes.join(', ')} FROM ` +
-                        `(SELECT ${topFragment}*` +
-                          ` FROM (SELECT ROW_NUMBER() OVER (ORDER BY ${orders.mainQueryOrder.join(', ')}) as row_num, * ` +
-                            ` FROM ${tables} AS ${tmpTable}${whereFragment})` +
-                          ` AS ${tmpTable} WHERE row_num > ${offset})` +
-                        ` AS ${tmpTable}`;
-        return fragment;
-      }
-      mainFragment = `SELECT ${topFragment}${attributes.join(', ')} FROM ${tables}`;
+    if (options.offset || isLimit) {
+      const offset = options.offset ? new Slot(options.offset) : '0';
+      composition.add('OFFSET ', offset, ' ROWS');
+      if (!isLimit) logger.warn('options.offset was used without options.limit. Most likely this was unintended.');
     }
 
-    if (mainTableAs) {
-      mainFragment += ` AS ${mainTableAs}`;
+    if (isLimit) {
+      composition.add(' FETCH NEXT ', new Slot(options.limit), ' ROWS ONLY');
     }
 
-    if (options.tableHint && TableHints[options.tableHint]) {
-      mainFragment += ` WITH (${TableHints[options.tableHint]})`;
-    }
-
-    return mainFragment;
+    return composition;
   }
 
-  addLimitAndOffset(options, model) {
-    // Skip handling of limit and offset as postfixes for older SQL Server versions
-    if (semver.valid(this.sequelize.options.databaseVersion) && semver.lt(this.sequelize.options.databaseVersion, '11.0.0')) {
-      return '';
-    }
-
-    const offset = options.offset || 0;
-    const isSubQuery = options.subQuery === undefined
-      ? options.hasIncludeWhere || options.hasIncludeRequired || options.hasMultiAssociation
-      : options.subQuery;
-
-    let fragment = '';
-    let orders = {};
-
-    if (options.order) {
-      orders = this.getQueryOrders(options, model, isSubQuery);
-    }
-
-    if (options.limit || options.offset) {
-      if (!options.order || options.include && !orders.subQueryOrder.length) {
-        fragment += options.order && !isSubQuery ? ', ' : ' ORDER BY ';
-        fragment += `${this.quoteTable(options.tableAs || model.name)}.${this.quoteIdentifier(model.primaryKeyField)}`;
-      }
-
-      if (options.offset || options.limit) {
-        fragment += ` OFFSET ${this.escape(offset)} ROWS`;
-      }
-
-      if (options.limit) {
-        fragment += ` FETCH NEXT ${this.escape(options.limit)} ROWS ONLY`;
-      }
-    }
+  booleanValue(value) {
+    return value ? 1 : 0;
+  }
 
-    return fragment;
+  bindParam(bind) {
+    return value => {
+      bind.push(value);
+      return `@${bind.length -1}`;
+    };
   }
 
-  booleanValue(value) {
-    return value ? 1 : 0;
+  outputBind(array) {
```

Comment:
An explanation as comment why MSSQL require this treatment, I know it is some format difference but please specify that difference so this method can be safely changed in future